sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -438,6 +439,17 @@ object CombineFilters extends Rule[LogicalPlan] {
 }
 
 /**
+ * Combines Limit and Sort if possible, so it will be planed as TakeOrdered
+ * and won't do total ordering.
+ */
+object CombineLimitSort extends Rule[LogicalPlan] {
```

Comment:
i think CombineLimitSort is more descriptive now, since this is single and special case of combine limit and sort. 
we can rename it in future when there are more push down cases we add here